storageClient/storageClient_test.go

```diff
@@ -270,6 +273,113 @@ func TestWriteFileWithBigMetadataKey(t *testing.T) {
 	}
 }
 
+func TestEncryptAndDecryptFileSingleBlock(t *testing.T) {
+	// create a plaintext file and file with a random string that is smaller than the blockSize
+	plainFile, err := os.Create("plaintext")
+	if err != nil {
+		t.Fatalf("Could not create plainFile: %+v", err)
+	}
+	randTxt, _ := e3dbClients.GenerateRandomString(500)
+	_, err = plainFile.WriteString(randTxt)
+	if err != nil {
+		t.Fatalf("Could not write to plainFile: %+v", err)
+	}
+	plainFile.Close()
+	ak := e3dbClients.RandomSymmetricKey()
+	// encrypt the file
+	_, _, err = e3dbClients.EncryptFile("plaintext", "encrypted", ak)
+	if err != nil {
+		t.Fatalf("Could not encrypt file: %s", err)
+	}
+	// decrypt the file
+	err = e3dbClients.DecryptFile("encrypted", "decrypted", ak)
+	if err != nil {
+		t.Fatalf("Could not decrypt file: %s", err)
+	}
+	// compare ptxt and decrypted
+	ptxt, err := ioutil.ReadFile("plaintext")
+	if err != nil {
+		t.Fatalf("Could not read plaintext file: %+v", err)
+	}
+	decrypted, err := ioutil.ReadFile("decrypted")
+	if err != nil {
+		t.Fatalf("Could not read decrypted file: %+v", err)
+	}
+	compare := bytes.Equal(ptxt, decrypted)
+	if !compare {
+		t.Fatalf("Plaintext and decrypted files do not match")
+	}
+	err = os.Remove("plaintext")
```

Comment:
for test either don't worry about (to the point of failing the test) not being able to remove the files and write
```
	plainFile, err := os.Create(plaintextFilename)
 	if err != nil {
 		t.Fatalf("Could not create %s: %+v", plaintextFilename , err)
 	}
      defer os.Remove(plaintextFilename)
```

or worry about logging the error but not failing the test with something like

```
	plainFile, err := os.Create(plaintextFilename)
 	if err != nil {
 		t.Fatalf("Could not create %s: %+v", plaintextFilename , err)
 	}
      defer func() {
       err:=   os.Remove(plaintextFilename)
      if err != nil {
 		t.Logf("Could not delete %s: %+v", plaintextFilename, err)
 	}
}()
```

also please move the filenames for plain, encrypted and decrypted up into variables block around line 30) and reuse those throughout the tests please.

Reply:
https://blog.golang.org/defer-panic-and-recover
https://tour.golang.org/flowcontrol/12
https://gobyexample.com/defer